nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/FetchGCSObject.java

```diff
@@ -273,7 +275,15 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
         final long millis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
         getLogger().info("Successfully retrieved GCS Object for {} in {} millis; routing to success", new Object[]{flowFile, millis});
-        session.getProvenanceReporter().fetch(flowFile, "https://" + bucketName + ".storage.googleapis.com/" + key, millis);
+
+        String transitUri;
+        try {
+            final URL url = new URL(storage.getOptions().getHost());
+            transitUri = String.format("%s://%s.%s/%s", url.getProtocol(), bucketName, url.getHost(), key);
+        } catch (MalformedURLException e) {
+            transitUri = e.getClass().getSimpleName();
+        }
```

Comment:
Using `URI.create()` avoids the checked `MalformedURLException` and because the Storage API URL must be valid for making the request, recommend the following approach:
```suggestion
        final URI storageApiUri = URI.create(storage.getOptions().getHost());
        final String transitUri = String.format("%s://%s.%s/%s", uri,getScheme(), bucketName, uri.getHost(), key);
```

Reply:
Yep, that's better.